tests/Model/CallbackIteratorTest.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace MongoDB\Tests\Model;
+
+use ArrayIterator;
+use MongoDB\Model\CallbackIterator;
+use MongoDB\Tests\TestCase;
+
+use function array_keys;
+use function iterator_to_array;
+use function strrev;
+
+class CallbackIteratorTest extends TestCase
+{
+    public function testArrayIteration(): void
```

Comment:
I'd propose rewriting all of these tests to use the following matrix:

* Callbacks in [various forms](https://www.php.net/manual/en/language.types.callable.php#language.types.callable.passing) that apply the same transformation (`*= 2` seems fine)
  * Closure
  * Object method (e.g. static method on this class)
* Iterables
  * ArrayIterator constructed from an indexed array
  * ArrayIterator constructed from an associative array (to show that keys are preserved)
  * IteratorAggregate for both of the above (per @GromNaN's [suggestion below](https://github.com/mongodb/mongo-php-library/pull/1126/files#r1255751584))

Using a data provider for iterables and expected arrays would probably be most efficient.

Reply:
@alcaeus I can make the changes if it frees you up.